### PR TITLE
[BugFix] use a tinyint col to fill empty project

### PR DESCRIPTION
**Pull request by packy92**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

This pr contains:
- use a tinyint col to fill empty project. This will add a extra cost of project at some scenes but decrease the volume of data when transfering them. 
- Add check for pushing agg to metaScan. Only `MAX`, `MIN`, `DICT_MERGE` agg func without group by clause can be pushed to metaScan, otherwise, it remains the agg->project->metaScan structure. When pushdown agg to the metaScan, the old project between agg and metaScan can be removed.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PruneProjectColumnsRule.java [hunk 1]
```diff
@@ -62,17 +63,10 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
...
-            ColumnRefOperator smallestColumn =
-                    inputOp.getSmallestColumn(candidates, context.getColumnRefFactory(), input.inputAt(0));
-            if (smallestColumn != null) {
-                ScalarOperator expr = projectOperator.getColumnRefMap().get(smallestColumn);
-                newMap.put(smallestColumn, expr);
-                requiredInputColumns.union(smallestColumn);
-            }
+        if (newMap.isEmpty()) {
+            ColumnRefOperator constCol = context.getColumnRefFactory()
+                    .create("const_col", Type.INT, false);
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PruneProjectColumnsRule.java:68]: Why `INT` ref but a `TINY_INT` const operator?

**Review by Seaven — DISMISSED** [review]

**Review by kangkaisen — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneProjectColumnsRule.java|4|4|100.00%|[]|

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by murphyatwork — DISMISSED** [review]

**Comment by packy92** [comment]
run starrocks_admit_test

**Comment by packy92** [comment]
run starrocks_admit_test

**Comment by packy92** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneProjectColumnsRule.java|4|4|100.00%|[]|

**Comment by packy92** [comment]
run starrocks_admit_test

**Comment by packy92** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneProjectColumnsRule.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownAggToMetaScanRule.java|1|1|100.00%|[]|

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PruneProjectColumnsRule.java [hunk 2]
```diff
@@ -62,17 +63,10 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
...
-        LogicalOperator inputOp = (LogicalOperator) input.inputAt(0).getOp();
-        if (newMap.isEmpty() && inputOp.getOpType().equals(OperatorType.LOGICAL_JOIN)) {
-            ColumnRefSet candidates = new ColumnRefSet(projectOperator.getColumnRefMap().keySet());
-            ColumnRefOperator smallestColumn =
-                    inputOp.getSmallestColumn(candidates, context.getColumnRefFactory(), input.inputAt(0));
-            if (smallestColumn != null) {
-                ScalarOperator expr = projectOperator.getColumnRefMap().get(smallestColumn);
-                newMap.put(smallestColumn, expr);
-                requiredInputColumns.union(smallestColumn);
-            }
```
Comment (LiShuMing) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PruneProjectColumnsRule.java:75]: Why not handle `PruneProjectRule`?  `getSmallestColumn` can be removed ?

**Comment by packy92** [comment]
run starrocks_fe_unittest

**Comment by packy92** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PruneProjectColumnsRule.java (hunk 2 above)
Comment (packy92) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PruneProjectColumnsRule.java:75]: If we have add the tinyint col when projectMap is empty in `PruneProjectColumnsRule`, I think the logic of filling smallest col in `PruneProjectRule` can be removed. I want to test this code for sometimes, if it's bugfree then remove the `PruneProjectRule`.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneProjectColumnsRule.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownAggToMetaScanRule.java|1|1|100.00%|[]|

**Review by kangkaisen — DISMISSED** [review]

**Comment by packy92** [comment]
run starrocks_admit_test

**Comment by packy92** [comment]
run starrocks_fe_unittest

**Review by Seaven — DISMISSED** [review]

**Review by kangkaisen — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by packy92** [comment]
run starrocks_fe_unittest

**Comment by packy92** [comment]
run starrocks_admit_test

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 20 / 20 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneProjectColumnsRule.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownAggToMetaScanRule.java|16|16|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**